src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -632,13 +632,13 @@ public static string GetClassNameFromInstance (IntPtr jobject)
 		}
 
 		[DllImport ("__Internal", CallingConvention = CallingConvention.Cdecl)]
-		internal static extern IntPtr monodroid_typemap_managed_to_java (string managed);
+		internal static extern IntPtr monodroid_typemap_managed_to_java (byte[] mvid, int token);
```

Comment:
Should we add more "error checking" to this interface, e.g. `mvid.Length`, so that native code doesn't read out-of-bounds?

Reply:
Not sure... it's internal, and I don't want to introduce any overhead. On the other hand, it would be a prudent thing to do

Reply:
The call is no longer internal, I added a wrapper function which is used by all the code that needs to use this.